integration/test/basic/main_test.go

```diff
@@ -0,0 +1,38 @@
+// +build k8srequired
+
+package basic
+
+import (
+	"testing"
+
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/certctl/integration/setup"
+)
+
+var (
+	f *framework.Host
+)
+
+// TestMain allows us to have common setup and teardown steps that are run
+// once for all the tests https://golang.org/pkg/testing/#hdr-Main.
+func TestMain(m *testing.M) {
+	var err error
+
+	l, err := micrologger.New(micrologger.Config{})
+	if err != nil {
+		panic(err.Error())
+	}
```

Comment:
The initialization here is a bit more explicit and also handled within the `init` function in other projects. I think it makes sense to comply with this.

Reply:
Done, PTAL.

Reply:
I do not understand why you just don't do copy/paste to keep the structure the same. Like this is now written differently and just harder to parse for humans when it is always different all the time. I do not want to bother with these things but it could way easier and I just don't understand this.

Reply:
Not sure I follow :) copy&paste is exactly what i've done, this is the commit https://github.com/giantswarm/certctl/pull/67/commits/8404d64f0fe3b78558df3be2419c8918feaed2a1 I also don't see how this lines can be hard to understand: https://github.com/giantswarm/certctl/blob/8404d64f0fe3b78558df3be2419c8918feaed2a1/integration/test/basic/main_test.go#L18-L34, maybe i'm missing something?

Reply:
Not sure where you copied from but the commit you show above is actually the code we talk about so it does not work as reference. When I look at the following tests I see they are consistent in initialization and the block style with config and constructure. 

- https://github.com/giantswarm/cluster-operator/blob/master/integration/test/basic/main_test.go
- https://github.com/giantswarm/aws-operator/blob/master/integration/test/clusterstate/main_test.go
- https://github.com/giantswarm/aws-operator/blob/master/integration/test/scaling/main_test.go
- https://github.com/giantswarm/azure-operator/blob/master/integration/test/clusterstate/main_test.go
- https://github.com/giantswarm/azure-operator/blob/master/integration/test/ready/main_test.go
- https://github.com/giantswarm/azure-operator/blob/master/integration/test/scaling/main_test.go

The ones that do not comply are these. 

- https://github.com/giantswarm/aws-operator/blob/master/integration/test/draining/main_test.go
- https://github.com/giantswarm/chart-operator/blob/master/integration/test/chartvalues/main_test.go
- https://github.com/giantswarm/chart-operator/blob/master/integration/test/basic/main_test.go

Reply:
Ok, I meant I copied&pasted the code from the `TestMain` function to a new `init` function. In this case we just need to instantiate a `host` type value, looks like the projects you link are more complex but good to know we are standardizing the setup, will keep in mind. Going with this, thx!

Reply:
The standardizing is exactly the point. When we get to have more code more equal we can abstract functionality once we see it and we only see it when it is rather more equal than not. So IMO our lives and our code will just be better when we try to be as consistent as possible and then it is easy for everybody to just catch up on something because you saw the same 15 lines in 4 different places and think about changing this now. And we can just apply this to anything, from plain lines of code to conceptual design ideas.